File: LeetCode/110. Balanced Binary Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public boolean isBalanced(TreeNode root) {
        return getHeight(root) != -1;
    }

    private int getHeight(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int lh = getHeight(root.left);
        if (lh == -1) {
            return -1;
        }
        int rh = getHeight(root.right);
        if (rh == -1) {
            return -1;
        }
        return Math.abs(lh - rh) > 1 ? -1 : Math.max(lh, rh) + 1;
    }
}

File: LeetCode/110. Balanced Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """

        def height(root):
            if not root:
                return 0
            lh = height(root.left)
            if lh == -1:
                return -1
            rh = height(root.right)
            if rh == -1:
                return -1
            if abs(lh - rh) > 1:
                return -1
            return max(lh, rh) + 1

        return height(root) != -1
